File: MaterialObligatorio/Implementacion/Codigo/Frontend/src/app/interfaces/product.ts
export interface Product {
  id: number;
  nombre: string;
  precio: number;
  descripcion: string;
  pharmacy: {
    id: number;
    name: string;  
  };
}

export class ProductRequest {
  nombre: string;
  precio: number;
  descripcion: string;
  pharmacyName: string = "";

  constructor(nombre: string, precio: number, descripcion: string, pharmacyName: string){
    this.nombre = nombre;
    this.precio = precio;
    this.descripcion = descripcion;
    this.pharmacyName = pharmacyName;
  }
}

File: MaterialObligatorio/Implementacion/Codigo/Backend/PharmaGo.Domain/Entities/Product.cs

namespace PharmaGo.Domain.Entities
{
    public class Product
    {
        public int Id { get; set; }
        public string Nombre { get; set; }

        public decimal Precio { get; set; }

        public string Descripcion { get; set; }

        public Pharmacy? Pharmacy { get; set; }
    }
}